mne_bids/write.py

```diff
@@ -120,12 +124,37 @@ def _channels_tsv(raw, fname, overwrite=False):
         ch_type.append(map_chs[_channel_type])
         description.append(map_desc[_channel_type])
     low_cutoff, high_cutoff = (raw.info['highpass'], raw.info['lowpass'])
-    if raw._orig_units:
+    if (
+        raw._orig_units is not None and
+        len(raw._orig_units) == len(raw.ch_names)
+    ):
         units = [raw._orig_units.get(ch, 'n/a') for ch in raw.ch_names]
```

Comment:
I don’t know why this line is not enough to deal with missing units in _orig_units and why you need to do this limbo below

Reply:
The handling we have in `main` relies on `raw._orig_units`, yes.

But I loaded an existing raw and appended the GSR and temperature channels via `raw.add_channels()`. This leads to a situation where `raw._orig_units` does **not** contain the two new channels (IDK if this is maybe a bug in MNE?)

So when I try to write this new raw, MNE-BIDS `main` discovers, "oh nice, there's `raw._orig_units`!` and uses those. But since the two newly-added channels are missing there, they'll get the unit `n/a` assigned, which is not correct.

This is not relevant for our tests here because we're loading the testing dataset and not modifying it "on the fly", so I can revert this change for now.

But there's certainly a bug here somewhere, either in MNE-BIDS, which shouldn't use `raw._orig_units`, or in MNE-Python, which should update `_orig_units` on `raw.add_channels()`

Reply:
For context, I've reverted the changes @agramfort complained about in [ece3ebf](https://github.com/mne-tools/mne-bids/pull/1059/commits/ece3ebff0221d731dfe49589dc1fa5034631ef23)

But we definitely need to look into this again, I very much believe we have a bug somewhere

Reply:
> or in MNE-Python, which should update _orig_units on raw.add_channels()

to me it's this one that should be fixed :point_up:

Reply:
@sappelhoff Could you maybe try to produce an MWE and open an issue with MNE? I'm rather busy right now with this PR and other things on my plate :(

Reply:
https://github.com/mne-tools/mne-python/issues/11119

Reply:
<3 thanks!!